FreeRTOS/FreeRTOS-Plus-TCP
# WIP - Separate out sockets

Separate out sockets, if the idea is approved then I will finish this PR and fix any affected unit tests.

Description
-----------
The sockets file has become the most cumbersome file, this separates TCP/UDP sockets and also better isolates socket code from other files. None of the API or logic is altered, this is mostly just copy pasta.

Test Steps
-----------
Not finished. Getting feedback first.

Related Issue
-----------
Size of FreeRTOS_Sockets.c file.


By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.
